### ISSUE TITLE
Error when switching between modules version (`--enable-versioning` option)

### ISSUE THREAD

**Issue by xdelaruelle** [cea-hpc/modules#531]
Modulefiles generated when `--enable-versioning` installation option is set do not allow to switch between the different installed version of Modules:

```console
$ ml 5.3.1
Loading modules
  ERROR: Conflicting 5.3.1 is loading

Loading 5.3.1
  ERROR: Load of requirement modules failed
```

**Comment by xdelaruelle** [comment]
Issue comes from the use of `module unload` command within the generated *versions* modulefiles. Since v4.2, use of `module unload` indicates a conflict toward module targeted for unload.

Adding the `--not-req` option to these `module unload` commands helps to not define a conflict toward these modules to unload.

Existing *versions* modulefiles for Modules version above 4.1 should be updated to add the `--not-req` option to the `module unload` commands. The following shell script should help to fix these modulefiles (replace `/usr/local/Modules/versions` by the location where the *versions* modulefiles are located on your setup):

```sh
for i in /usr/local/Modules/versions/*; do
    version=$(basename "$i")
    version=$i
    major=${version:0:1}
    minor=${version:2:1}
    if [[ "$major" -gt 4 || ("$major" -eq 4 && "$minor" -gt 1) ]]; then
        sed -i "s|module unload|module unload --not-req|" "$i"
    fi
done
```

**Comment by xdelaruelle** [comment]
Fixed by [58476f8](https://github.com/cea-hpc/modules/commit/58476f81177309b364ac2d48d19a3eddf7b6a572)
